app/utils/importFile.js

```diff
@@ -16,35 +16,55 @@ const importFile = async ({
   documentId,
   collectionId,
 }: Options): Promise<Document> => {
-  return new Promise((resolve, reject) => {
-    const reader = new FileReader();
+  return new Promise(async (resolve, reject) => {
+    // non plain text support
+    if (documents.importFiletypes.includes(file.type)) {
+      try {
+        const document = new Document(
+          {
+            parentDocumentId: documentId,
+            collectionId,
+            title: file.name.replace(/\.[^/.]+$/, ""),
+          },
+          documents
+        );
```

Comment:
This last bit of `Document instantiation is the only thing that's throwing me right now – can we just pass these params all through directly to the `import` store method?

Reply:
https://github.com/outline/outline/pull/1469/commits/44a861b3af5b5db96d9a0a15c042bda85c2f2b17#diff-cf91b760df0750438c333d13b1499242R470